chiselFrontend/src/main/scala/chisel3/Bits.scala

```diff
@@ -1830,216 +1807,193 @@ package experimental {
+    }
+
+    override def do_>> (that: Int)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Interval = {
+      binop(sourceInfo, Interval(this.range >> that), ShiftRightOp, that)
     }
-    override def do_>> (that: Int)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Interval =
-      binop(sourceInfo,
-        Interval(this.width.shiftRight(that), this.range >> that), ShiftRightOp, that)
 
     override def do_>> (that: BigInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Interval =
       do_>>(that.toInt)
 
     override def do_>> (that: UInt)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Interval = {
       // Worst case range is when you shift by nothing
-      val newRange = this.range
-      binop(sourceInfo, Interval(this.width, newRange), DynamicShiftRightOp, that)
+      val newRange = IntervalRange(firrtlir.UnknownBound, firrtlir.UnknownBound, this.range.binaryPoint)
+      binop(sourceInfo, Interval(newRange), DynamicShiftRightOp, that)
     }
 
     /**
-      * Wrap the value of this [[Interval]] into the range of a different Interval with a presumably smaller range.
-      * @param that
-      * @return
-      */
-    final def wrap(that: Interval): Interval = macro SourceInfoTransform.thatArg
-
-    def do_wrap(that: Interval)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Interval = {
-      (that.range.lowerBound, that.range.upperBound) match {
-        case (lower: firrtlconstraint.IsKnown, upperBound: firrtlconstraint.IsKnown) =>
-        // things are good, we have known ranges
-        case _ =>
-          throwException("wrap requires an Interval argument with known lower and upper bounds")
-      }
-      val dest = Interval(IntervalRange(that.range.lowerBound, that.range.upperBound, this.range.binaryPoint))
-      val other = that
-      requireIsHardware(this, s"'this' ($this)")
-      requireIsHardware(other, s"'other' ($other)")
-      pushOp(DefPrim(sourceInfo, dest, WrapOp, this.ref, other.ref))
-    }
-
-    /**
-      * Squeeze the value of this [[Interval]] into the range of a different Interval with a presumably smaller range.
+      * Squeeze the value of this [[Interval]] into the intersection of the ranges.
       * @param that
       * @return
       */
     final def squeeze(that: Interval): Interval = macro SourceInfoTransform.thatArg
 
     def do_squeeze(that: Interval)(implicit sourceInfo: SourceInfo, compileOptions: CompileOptions): Interval = {
-      (that.range.lowerBound, that.range.upperBound) match {
-        case (lower: firrtlconstraint.IsKnown, upperBound: firrtlconstraint.IsKnown) =>
-        // things are good, we have known ranges
-        case _ =>
-          throwException("squeeze requires an Interval argument with known lower and upper bounds")
-      }
       val dest = Interval(IntervalRange(that.range.lowerBound, that.range.upperBound, this.range.binaryPoint))
```

Comment:
This isn't correct - as mentioned below, squeeze return range has a lower bound which is the larger of the two input lower bounds, and an upper bound which is the smaller of the two input upper bounds.